packages/ledger/src/index.ts

```diff
@@ -0,0 +1,270 @@
+    }
+  }
+}
+
+const getAddresses = (
+  account: LedgerAccount,
+  asset: Asset,
+  currentChain: Chain,
+  offset: number = 0,
+  limit: number = 15
+): Promise<Account[]> =>
+  limit - offset <= 0
+    ? (() => {
+        throw new Error('Offset must be less than limit')
+      })()
+    : Promise.all(
+        Array.from({ length: limit - offset }, (_, index) =>
+          getAddress(account, asset, currentChain, index)
+        )
+      )
+
+function ledger(): WalletInit {
+  return () => {
+    let accounts: Account[] | undefined
+    return {
+      label: 'Ledger',
+      getIcon: async () => (await import('./icon')).default,
+      getInterface: async ({ EventEmitter, chains }) => {
+        const Eth = (await import('@ledgerhq/hw-app-eth')).default
+        const { TransactionFactory: Transaction, Capability } = await import(
+          '@ethereumjs/tx'
+        )
+        const { default: Common, Hardfork } = await import('@ethereumjs/common')
+        const { rlp } = (await import('ethereumjs-util')).default
+
+        const transport: Transport = await getTransport()
+        const eth = new Eth(transport)
+        const eventEmitter = new EventEmitter()
+
+        let currentChain: Chain | undefined
+        const scanAccounts = async ({
+          derivationPath,
+          chainId,
+          asset
+        }: ScanAccountsOptions): Promise<Account[]> => {
+          currentChain = chains.find(({ id }) => id === chainId)
+          // TODO: Maybe throw error here
+          if (!currentChain) return []
+
+          const { publicKey, chainCode } = await eth.getAddress(
+            derivationPath,
+            false,
+            true // set to true to return chainCode
+          )
+
+          const { compress } = (await import('eth-crypto')).publicKey
+
+          return getAddresses(
+            {
+              publicKey: compress(publicKey),
+              chainCode: chainCode ?? '',
+              derivationPath: derivationPath
+            },
+            asset,
+            currentChain
+          )
+        }
+
+        const getAccounts = async () =>
+          accounts ??
+          (accounts = await accountSelect({
+            basePaths: DEFAULT_BASE_PATHS,
+            assets,
+            chains,
+            scanAccounts,
+            walletIcon: '<svg></svg>'
+          }))
+
+        const ledgerProvider = {}
+
+        const provider = createEIP1193Provider(ledgerProvider, {
+          eth_requestAccounts: async baseRequest => {
+            // Triggers the account select modal if no accounts have been selected
+            const [{ address }] = await getAccounts()
+            return [address]
+          },
+          eth_accounts: async baseRequest => {
+            return accounts?.[0]?.address ? [accounts[0].address] : []
+          },
+          eth_chainId: async baseRequest => {
+            console.log('eth_chainId called', currentChain?.id)
+            return currentChain?.id ?? ''
```

Comment:
@aaronbarnardsound I am noticing this method get's called before the `scanAccounts` method so the chainId isn't set yet -- is this desired behaviour?

Reply:
This should be fixed in the latest commits